apps/api/src/server/applications/application/application.validators.js

```diff
@@ -55,6 +56,20 @@ const validateExistingApplication = async (value) => {
 	}
 };
 
+/**
+ *
+ * @param {number} value
+ * @throws {Error}
+ * @returns {Promise<void>}
+ */
+const validateExistingRepresentation = async (value) => {
+	const representation = await representationRepository.getById(value);
+
+	if (representation === null || typeof representation === 'undefined') {
+		throw new Error('Unknown representation');
+	}
+};
```

Comment:
So we have been moving away from making DB calls in the validators that we use before the api call reaches the controller. As such, we are validating the inputs at this stage to check that they exist and are in the right format. Checking if a representation is in the DB would then happen at the controller level. This is essentially because we came to an agreement that we would like to not make any duplicate API calls and with this approach, we can pass around responses from databases within the controller more easily.

Reply:
This is following the patterns in place. Similar to -all the other functions. 

Could you provide an example where you have moved away from this pattern and if all the validators have been refactored to the 'one api call only' approach. 

Would we need to change this single instance over or all of the functions in the file ? 

Feels like it should be addressed as one tech debt ticket.

Reply:
I understand the confusion and I would have done the same in your position. An example I can provide is the endpoint in `apps/api/src/server/applications/application/documents/document.routes.js` on line 134 (`getDocumentProperties`). Within the `getDocumentProperties` controller we check if the document exists in the DB and throw an error if it does not. 

We are not entirely in a position to spend a lot of time addressing tech debt at this point, though it would be a discussion we could have with @francescobrenci-AR.

In essence, I agree that it would be idea to address instances of this across the codebase, but I am not certain we are in a position to do so.